testing/fetcher.ts

```diff
@@ -322,11 +324,20 @@ export class AuthenticatingFetcher implements Fetcher {
           }
         });
 
+        const newHeaders: FetchRequest['headers'] = headersWithSubstitutions ? JSON.parse(headersWithSubstitutions) : undefined;
+        if (newHeaders && digestTransform) {
+          const {algorithm, header} = digestTransform;
+          const headerValue = newHeaders[header];
+          if (headerValue) {
+            newHeaders[header] = createHash(algorithm).update(headerValue, 'utf-8').digest('hex')
```

Comment:
I'd suggest making the digest encoding variable as well, to make this a bit more flexible.